Adds command to set the coordinates of portals

froyo4u opened the pull request:
Adds the command: `/portal set_portal_position`.

froyo4u committed:
Merge pull request #1 from qouteall/1.16

Merge changes from qouteall

froyo4u committed:
Merge pull request #2 from qouteall/1.16

Merge Changes

froyo4u committed:
Added the ability to configure the Dimension Stack Divider

Do note: It requires restarting the game to take effect.

froyo4u committed:
Update en_us.json

froyo4u committed:
Merge pull request #3 from qouteall/1.16

Merge Changes

froyo4u committed:
Added the Location Command

Adds a command: /portal set_portal_position

froyo4u committed:
Merge pull request #4 from MeowMC/1.16

Push Previous Changes

froyo4u committed:
Revert "Push Previous Changes"

froyo4u committed:
Merge pull request #5 from MeowMC/revert-4-1.16

Revert "Push Previous Changes"

qouteall commented:
Why do you want to add "originPos1" into every portal?

MeowMC commented:
> Why do you want to add "originPos1" into every portal?

I am still using the original variable `origin`, I just had to create new methods in order to achieve what I wanted because the original method `getOriginPos` was already being used for many other things and I didn't want to mess with it. If you would like I could rename the methods and variables or you could do it. In my testing, however, it works fine with no issues(at least none I've encountered).

qouteall commented:
It's so weird. Why do you want to create a new field that has the same meaning as the portal's position? Why can't you just `setPosition`?
> I am still using the original variable origin


There is no "original variable origin". It has position

> I just had to create new methods in order to achieve what I wanted because the original method getOriginPos was already being used for many other things and I didn't want to mess with it

The origin position is the portal entity's position. Don't try to "avoid messing" things because if the mechanics is not understood more things will be messed. If you don't understand one thing, ask me.

froyo4u committed:
Remove dimension arguement

froyo4u committed:
Updated methods and variables

`getOriginPos1` was removed, and `setOriginPos1` was changed to `setOriginPos`. The variable `origin` was removed and in the method `setOriginPos` I changed the variable `origin` to `pos`.

MeowMC commented:
I updated the method and variable names to match what you said previously. Also if you're wondering `setOriginPos` actually executes `setPos`.

qouteall commented:
1. The `set_portal_destination` accepts firstly a dimension argument then the position argument. But this command does not accept the dimension argument. This imparity may be confusing.
2. Why do you add the y level by 1 in `setOriginPos`? This behavior makes the method name misleading and may induce potential bugs. You can add the y by 1 outside of the method.
3. Other places use `updatePosition` to set the portal position. The difference between `setPos` and `updatePosition` is that the second updates the bounding box. I am going to add a new method for setting the position but that method also includes invalidating the cache including the bounding box. You can simply inline the `setOriginPos` method.
4. The code of displaying warning message gets repeated. The repeated code should be wrapped in another method.

MeowMC commented:
Sorry, I forgot to mention but I did actually remove the dimension argument as I realized the method was `dimensionTo`.

MeowMC commented:
I will work on the other things you have mentioned.

froyo4u committed:
Merge pull request #6 from qouteall/1.16

Merge changes from qouteall

froyo4u committed:
Made changes to setOriginPos

I inlined `setOriginPos`

MeowMC commented:
What code should be wrapped in another method?

qouteall commented on imm_ptl_core/src/main/java/com/qouteall/immersive_portals/commands/PortalCommand.java:
```diff
@@ -1312,6 +1324,34 @@ private static void invokeSetPortalDestination(
+
+        if (portal instanceof BreakablePortalEntity) {
+            if (!((BreakablePortalEntity) portal).unbreakable) {
+                sendMessage(context, "You are editing a breakable portal." +
```
This warning can be wrapped into another method. You can also remove it because the warning is not mandatory

qouteall commented on imm_ptl_core/src/main/java/com/qouteall/immersive_portals/commands/PortalCommand.java:
```diff
@@ -1312,6 +1324,34 @@ private static void invokeSetPortalDestination(
+                context, "pos"
+                );
+        double posX = pos.getX();
+        double posY = pos.getY() + 1.000;
```
Well I think that the command user may still be confused by its y level adding by 1. Maybe this adding by 1 can be removed.

froyo4u replied:
I was thinking that too as other commands like `set_portal_destination` do not do this so it will probably confuse them.

qouteall commented:
You can use `ServerTeleportationManager.teleportEntityGeneral` to teleport entities across dimension. This operation may create a new entity object

MeowMC commented:
> You can use `ServerTeleportationManager.teleportEntityGeneral` to teleport entities across dimension. This operation may create a new entity object

Will do

froyo4u committed:
Added stuff and removed stuff

I removed the setpos and used `ServerTeleportationManager.teleportEntityGeneral` instead, this seems to work. I also removed the warning as you said it was not needed.

froyo4u committed:
Merge pull request #7 from qouteall/1.16

Merge Changes

MeowMC commented:
Did the stuff you said 👍🏻
